flocker/common/_retry.py

```diff
@@ -175,3 +178,39 @@ def poll_until(predicate, steps, sleep=None):
     if result:
         return result
     raise LoopExceeded(predicate, result)
+
+
+def retry_effect_with_timeout(effect, timeout, retry_wait=1, exp_backoff=True,
+                              time=time.time):
+    """
+    If ``effect`` fails, retry it until ``timeout`` expires.
+
+    To avoid excessive retrying, this function by default (it can be disabled)
+    uses the exponential backoff algorithm, waiting twice the time between
+    each retry.
```

Comment:
Thanks for the helpful comment. Minor English points:

* Drop "(it can be disabled)", since it's implied by "by default"
* s/twice/double/ (I think twice is technically correct, but it's weird)
* move "by default" to after "algorithm"